http://sources.redhat.com/ml/libc-alpha/2009-09/msg00025.html

2009-09-13  Samuel Thibault  <samuel.thibault@ens-lyon.org>

        * hurd/hurdchdir.c (_hurd_change_directory_port_from_name):
        Return ENOENT when name is empty.
        * sysdeps/mach/hurd/chroot.c (chroot): Return ENOENT when path
        is empty.

---
 hurd/hurdchdir.c           |    6 ++++++
 sysdeps/mach/hurd/chroot.c |    6 ++++++
 2 files changed, 12 insertions(+)

--- a/hurd/hurdchdir.c
+++ b/hurd/hurdchdir.c
@@ -38,6 +38,12 @@
   len = strlen (name);
   if (len >= 2 && name[len - 2] == '/' && name[len - 1] == '.')
     lookup = name;
+  else if (len == 0)
+    {
+      /* Special-case null pathname according to POSIX */
+      errno = ENOENT;
+      return -1;
+    }
   else
     {
       char *n = alloca (len + 3);
--- a/sysdeps/mach/hurd/chroot.c
+++ b/sysdeps/mach/hurd/chroot.c
@@ -38,6 +38,12 @@
   len = strlen (path);
   if (len >= 2 && path[len - 2] == '/' && path[len - 1] == '.')
     lookup = path;
+  else if (len == 0)
+    {
+      /* Special-case null pathname according to POSIX */
+      errno = ENOENT;
+      return -1;
+    }
   else
     {
       char *n = alloca (len + 3);
